cumulus/parachains/runtimes/assets/asset-hub-westend/src/xcm_config.rs

```diff
@@ -349,6 +351,33 @@ pub type TrustedTeleporters = (
 	IsForeignConcreteAsset<FromSiblingParachain<parachain_info::Pallet<Runtime>>>,
 );
 
+/// Asset converter for trust-backed assets.
+/// Used to convert assets marked as `sufficient` into the asset needed for fee payment.
+/// This type allows paying fees in `sufficient` trust backed-assets.
+pub type TrustBackedSufficientAssetsConverter = SufficientAssetConverter<
+	Runtime,
+	TrustBackedAssetsConvertedConcreteId,
+	pallet_assets::BalanceToAssetBalance<Balances, Runtime, ConvertInto, TrustBackedAssetsInstance>,
+	TrustBackedAssetsInstance,
+>;
+
+/// Asset converter for foreign assets.
+/// Used to convert assets marked as `sufficient` into the asset needed for fee payment.
+/// This type allows paying fees in `sufficient` foreign assets.
+pub type ForeignSufficientAssetsConverter = SufficientAssetConverter<
+	Runtime,
+	ForeignAssetsConvertedConcreteId,
+	pallet_assets::BalanceToAssetBalance<Balances, Runtime, ConvertInto, ForeignAssetsInstance>,
+	ForeignAssetsInstance,
+>;
+
+/// Asset converter for pool assets.
+/// Used to convert assets in pools to the asset required for fee payment.
+/// The pool must be between the first asset and the one required for fee payment.
+/// This type allows paying fees with any asset in a pool with the asset required for fee payment.
+// TODO: Finish.
+pub type PoolAssetsConverter = ();
```

Comment:
Looks like you are on the right track doing this. With respect to `*SufficientAssetsConverter`, the idea in the [Trader](https://github.com/paritytech/polkadot-sdk/blob/da83c684e9c04dc3a0dfbd36bb7d50b15fe7c75f/cumulus/parachains/runtimes/assets/asset-hub-westend/src/xcm_config.rs#L400) was to remove `TakeFirstAssetTrader` and keep `SwapFirstAssetTrader`. They were both kept in for one release cycle because there is some time required for the pools to form and we didn't want sufficient asset support to break.

Basically, the intent is that asset sufficiency be related only to account existence and be decoupled from fee payment. *Any* asset should be equal for fee payment and use the "real" value through the pool, not the "fixed" value we use for sufficient ones.

Reply:
Well, one release cycle of Polkadot/Kusama, obviously many of Westend 😬 . We should probably remove them in another PR when we do for Polkadot.

Reply:
Sounds good, I'll make a test working with pools. We can easily have these `*SufficientAssetsConverter` in the fellowship runtimes or just go with the pools one.